lib/src/main/kotlin/com/swmansion/starknet/service/http/OkHttpService.kt

```diff
@@ -12,9 +12,13 @@ import java.io.IOException
 import java.util.concurrent.CompletableFuture
 
 /**
- * Service for making http requests.
+ * Service for making http requests using OkHttp library. You can reuse provide it with your client to
+ * <a href="https://square.github.io/okhttp/4.x/okhttp/okhttp3/-ok-http-client/#okhttpclients-should-be-shared"> avoid
+ * wasting resources</a>.
+ * @param client OkHttpClient used for making requests
  */
-internal class OkhttpHttpService() : HttpService {
+class OkHttpService(private val client: OkHttpClient) : HttpService {
+    constructor() : this(OkHttpClient())
```

Comment:
Perhaps we shouldn't be providing this secondary constructor at all if we want users to reuse the clients?

Reply:
This way they can reuse http service.

Reply:
I mean the constructor where `OkHttpClient` is automatically created, not the primary one.

Reply:
Default construtor -> reuse HttpService
Constructor with client -> reuse OkHttpClient